.github/workflows/publish.yaml

```diff
@@ -0,0 +1,213 @@
+# input. The PAT is necessary to push to your BCR fork as well as to open up a pull request against a registry.
+# At the moment, fine-grained PATs are not supported because they cannot open pull requests against public 
+# repositories, although this is on GitHub's roadmap: https://github.com/github/roadmap/issues/600.
+#
+# The module repository must contain a .bcr folder containing Publish to BCR templates.
+# See https://github.com/bazel-contrib/publish-to-bcr/tree/main/templates.
+#
+# Repositories containing multiple modules that are versioned together will have all modules included in
+# the published entry. This is controlled via the `moduleRoots` property in .bcr/config.yml.
+
+on:
+  # Make this workflow reusable, see
+  # https://github.blog/2022-02-10-using-reusable-workflows-github-actions
+  workflow_call:
+    inputs:
+      tag_name:
+        required: true
+        description: The tag identifying the release the publish to a Bazel registry.
+        type: string
+      registry_fork:
+        required: true
+        description: The Bazel registry fork to push to when opening up a pull request, e.g. "mycompany/bazel-central-registry"
+        type: string
+      registry:
+        description: The Bazel registry to open up a pull request against. Defaults to the Bazel Central Registry.
+        default: 'bazelbuild/bazel-central-registry'   
+        type: string
+      repository:
+        description: The Bazel module repository to publish an entry for. Defaults the the repository the action runs in.
+        default: ${{ github.repository }}
+        type: string
+      registry_branch:
+        description: The branch of the Bazel registry to open a PR against. Defaults to main.
+        default: main
+        type: string
+    secrets:
+      publish_token:
+        required: true
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout the module repository
+      uses: actions/checkout@v4.2.2
+      with:
+        ref: ${{ github.event.inputs.tag_name }}
+        repository: ${{ inputs.repository }}
+        path: this
+
+    - name: Checkout BCR
+      uses: actions/checkout@v4.2.2
+      with:
+        repository: ${{ github.event.inputs.registry }}
+        token: ${{ secrets.GITHUB_TOKEN }}
+        path: bazel-central-registry
+
+    # Get version from the tag, stripping any v-prefix
+    - name: Write release version
+      env:
+        TAG: ${{ inputs.tag_name }}
+      run: |
+        VERSION=${TAG#v}
+        echo Version: $VERSION
+        echo "VERSION=$VERSION" >> $GITHUB_ENV
+
+    # Create an initial entry so that we can attest the generated source.json and MODULE.bazel
+    # files. These are needed for the attestations.json template (chicken-and-egg) which is included
+    # in the entry that we will publish later. This entry will be discarded.
+    - name: Create entry
+      id: create-entry
+      uses: bazel-contrib/publish-to-bcr@fa7d3c8ad241c2c0cde639eb2225be55816e9565
```

Comment:
TODO: cut a release of the publish to bcr action and point to the version tag

Reply:
I think a spare checkout here won't work well because we want to checkout particular directories under `modules`, however we don't know the names of the modules we're publishing until the publish-to-bcr create entry runs. The module names are taken from the downloaded release archives' MODULE.bazel files.

Reply:
yes - but - no one should use the version tag if they care about supply chain security since there's no guarantee that tags are immutable